File: spring-reactive-02/learn-java/src/main/java/boki/learnjava/v2_filtersequence/a_filter/FilterEx1.java
package boki.learnjava.v2_filtersequence.a_filter;

import boki.learnjava.utils.Logger;
import reactor.core.publisher.Flux;

/**
 * Filter 기본 개념 예제
 */
public class FilterEx1 {
    public static void main(String[] args) {
        Flux
            .range(1, 20)
            .filter(num -> num % 2 == 0)
            .subscribe(Logger::onNext);
    }
}

File: spring-reactive-02/learn-kt/src/main/kotlin/boki/learnkt/v2_filtersequence/a_filter/FilterEx1.kt
package boki.learnkt.v2_filtersequence.a_filter

import boki.learnkt.util.Logger
import reactor.core.publisher.Flux

/**
 * Filter 기본 개념 예제
 */
fun main() {
    Flux
        .range(1, 20)
        .filter { it % 2 == 0 }
        .subscribe { Logger.onNext(it) }
}